cmd/principal/main.go

```diff
@@ -275,6 +280,9 @@ func NewPrincipalRunCommand() *cobra.Command {
 	command.Flags().BoolVar(&enableResourceProxy, "enable-resource-proxy",
 		env.BoolWithDefault("ARGOCD_PRINCIPAL_ENABLE_RESOURCE_PROXY", true),
 		"Whether to enable the resource proxy")
+	command.Flags().DurationVar(&waitDurationForAgentPing, "wait-duration-for-agent-ping",
+		env.ParseDuration(env.StringWithDefault("ARGOCD_PRINCIPAL_WAIT_DURATION_FOR_AGENT_PING", nil, "")),
```

Comment:
Please take a look at my comment for the new function in the `env` package to see what should be changed here.